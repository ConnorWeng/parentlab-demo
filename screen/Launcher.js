import React, { Component } from 'react';
import { StyleSheet, Text, View, Image, ImageBackground, Dimensions } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import PrimaryButton from '../component/PrimaryButton';

export default class Launcher extends Component {
  static navigationOptions = {
    header: null,
  }

  render() {
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.titleContainer}>
          <Text style={styles.title}>Grow Together!</Text>
        </View>
        <View style={styles.imageContainer}>
          <Image style={styles.biking} source={require('../resource/biking.png')}/>
        </View>
        <View style={styles.footerContainer}>
          <PrimaryButton style={styles.button} label="Get Started" onPress={() => this.props.navigation.navigate('GetActive')}/>
            <View style={styles.copyrightsContainer}>
              <Text style={styles.copyrights}>By continuing, you agree to our </Text>
              <Text style={styles.terms}>Terms of Use & Privacy Policy</Text>
            </View>
        </View>
      </SafeAreaView>
    );
  }
}

const { height, width } = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  titleContainer: {
    flex: 1,
  },
  imageContainer: {
    flex: 7,
    flexDirection: 'column',
    height: height/2,
  },
  buttonContainer: {
    flex: 1,
    width: width,
  },
  footerContainer: {
    flex: 2,
    alignItems: 'center',
    justifyContent: 'space-between',
    width: width,
  },
  button: {
    justifyContent: 'center',
  },
  biking: {
    width: width * 1.2,
    resizeMode: Image.resizeMode.contain,
  },
  title: {
    fontSize: 28,
    paddingTop: 40,
  },
  copyrightsContainer: {
    flexDirection: 'row'
  },
  copyrights: {
    fontSize: 10,
    color: 'rgba(0, 0, 0, 0.2)'
  },
  terms: {
    fontSize: 10,
    color: 'rgba(0, 0, 0, 0.5)'
  },
});
