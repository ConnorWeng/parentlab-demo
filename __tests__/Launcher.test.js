import React from 'react';
import Launcher from '../screen/Launcher';

import renderer from 'react-test-renderer';

jest.mock('react-redux', () => {
  return {
    connect: () => (component) => component
  };
});

it('renders correctly', () => {
  const rendered = renderer.create(<Launcher/>).toJSON();
  expect(rendered).toMatchSnapshot();
});
