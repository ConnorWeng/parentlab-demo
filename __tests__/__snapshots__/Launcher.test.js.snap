// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "paddingTop": 40,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 28,
          },
          Object {
            "paddingTop": 40,
          },
        ]
      }
    >
      Grow Together!
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 7,
        "flexDirection": "column",
        "height": 667,
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../resource/ground.png",
          }
        }
        style={
          Object {
            "resizeMode": undefined,
          }
        }
      />
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../resource/biking.png",
        }
      }
      style={
        Object {
          "resizeMode": "contain",
          "width": 900,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D9F1F5",
        "flex": 2.5,
        "justifyContent": "space-between",
        "paddingBottom": 40,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#43BCD3",
            "borderRadius": 20,
            "height": 42,
            "justifyContent": "center",
            "opacity": 1,
            "paddingLeft": 80,
            "paddingRight": 80,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
          Get Started
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.2)",
            },
          ]
        }
      >
        By continuing, you agree to our 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "rgba(0, 0, 0, 0.5)",
            },
          ]
        }
      >
        Terms of Use & Privacy Policy
      </Text>
    </View>
  </View>
</View>
`;
